main.go

```diff
@@ -345,14 +350,30 @@ func main() {
 		setup(ctx, config, db)
 	} else {
 		router.Use(func(ctx *gin.Context) {
-			if ctx.Request.Method != "OPTIONS" {
+			if ctx.Request.Method == "OPTIONS" {
+				ctx.Next()
+				return
+			}
+
+			// Check if it's a WebSocket upgrade request
+			if isWebSocketRequest(ctx.Request) {
+				token := ctx.Query("token")
```

Comment:
**REALLY BAD** : This creates a token leak.

Reply:
swapped to protocol header